File: sum.py
import json
from flask import Flask, request, jsonify, make_response

app = Flask(__name__)

@app.route('/sum', methods=['POST'])
def calculate_sum():
    try:
        # Get JSON data from the request
        data = request.get_json()

        # Extract 'a' and 'b' values
        a = int(data.get('a', 0))
        b = int(data.get('b', 0))

        # Calculate the sum
        result = a + b

        # Prepare the result data to be saved in JSON file
        output_data = {
            "a": a,
            "b": b,
            "sum": result
        }

        # Prepare response and set file download headers
        response = make_response(jsonify(output_data))
        response.headers['Content-Disposition'] = 'attachment; filename=result.json'
        response.headers['Content-Type'] = 'application/json'

        # Return response to be downloaded as result.json
        return response
    except Exception as e:
        return jsonify({"error": str(e)})

if __name__ == '__main__':
    app.run(host="0.0.0.0", port=5000)  # 0.0.0.0 makes it accessible externally

File: sum.php
<?php
// Get JSON input from frontend (index.html)
$data = json_decode(file_get_contents("php://input"), true);

// Check if values 'a' and 'b' are provided
if (!isset($data['a']) || !isset($data['b'])) {
    echo json_encode(["error" => "Missing parameters"]);
    exit;
}

// Prepare data to send to Flask API
$postData = json_encode([
    "a" => intval($data['a']),
    "b" => intval($data['b'])
]);

// Initialize cURL request to Flask API (running on your server IP)
$ch = curl_init("http://15.207.6.165:5000/sum");  // URL of Flask API
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);  // Get response back
curl_setopt($ch, CURLOPT_POST, true);  // POST request
curl_setopt($ch, CURLOPT_POSTFIELDS, $postData);  // Data to send
curl_setopt($ch, CURLOPT_HTTPHEADER, ['Content-Type: application/json']);  // Set content type

// Execute the request and capture the response
$response = curl_exec($ch);
curl_close($ch);

// Return JSON response (will be a downloadable file)
header("Content-Type: application/json");
header("Content-Disposition: attachment; filename=result.json");
echo $response;
?>
